ansible/condor.yml

```diff
@@ -45,58 +45,72 @@
         url: https://get.htcondor.org
         dest: /root/condor_install.sh
         mode: "0755"
-
-    - ansible.builtin.user:
-        user: clauncher
-        create_home: yes
-        append: true
-        groups: docker
   tags:
     - worker
     - manager
     - submission
 
-- name: install the central manager
+- name: install condor
   become: true
-  hosts: condor_manager
+  hosts: condor
   tasks:
+    - ansible.builtin.set_fact:
+        install_command:
+          - "./condor_install.sh"
+          - "--no-dry-run"
+          - "--password"
+          - "{{ condor_pool_password }}"
+        cm_opts:
+          - "--central_manager"
+          - "{{ groups['condor_manager'][0] }}"
+        exec_opts:
+          - "--execute"
+          - "{{ groups['condor_manager'][0] }}"
+        submit_opts:
+          - "--submit"
+          - "{{ groups['condor_manager'][0] }}"
+
+    - ansible.builtin.set_fact:
+        install_command: "{{ install_command + cm_opts }}"
+      when: "'condor_submit' in group_names"
+
+    - ansible.builtin.set_fact:
+        install_command: "{{ install_command + exec_opts }}"
+      when: "'condor_worker' in group_names"
+
+    - ansible.builtin.set_fact:
+        install_command: "{{ install_command + submit_opts }}"
+      when: "'condor_submit' in group_names"
+
     - ansible.builtin.shell:
-        cmd: ./condor_install.sh --no-dry-run --password "{{ condor_pool_password }}" --central-manager "{{ groups['condor_manager'][0] }}"
+        cmd: "{{ install_command }}"
         chdir: "/root/"
         creates: "/usr/sbin/condor_master"
+
+    - ansible.builtin.import_role:
+        name: condor_configuration
```

Comment:
Building the command based on the node group and executing it allows a single node to have multiple roles.